can't use mypy's warn_unused_ignores flag

Adding `warn_unused_ignores = True` to `mypy.ini` creates mypy errors, depending on the Python version used.

With Python 3.6:
```
$ mypy porcupine/ more_plugins/
porcupine/utils.py:21: error: unused 'type: ignore' comment
Found 1 error in 1 file (checked 56 source files)
```

With Python 3.7:
```
$ mypy porcupine/ more_plugins/
porcupine/utils.py:622: error: unused 'type: ignore' comment
porcupine/utils.py:627: error: unused 'type: ignore' comment
porcupine/utils.py:632: error: unused 'type: ignore' comment
Found 3 errors in 1 file (checked 56 source files)
```

--- a/mypy.ini
+++ b/mypy.ini
@@ -1,5 +1,6 @@
 [mypy]
 warn_unused_configs = True
+warn_unused_ignores = True
 disallow_untyped_defs = True
 show_error_codes = True
 custom_typeshed_dir = typeshed

--- a/porcupine/plugins/langserver.py
+++ b/porcupine/plugins/langserver.py
@@ -25,7 +25,7 @@ from typing import IO, Dict, List, NamedTuple, Optional, Tuple, Union, cast
 if platform.system() != 'Windows':
     import fcntl
 
-import sansio_lsp_client as lsp  # type: ignore[import]
+import sansio_lsp_client as lsp
 
 from porcupine import get_tab_manager, tabs, textwidget, utils
 from porcupine.plugins import autocomplete, underlines

--- a/porcupine/utils.py
+++ b/porcupine/utils.py
@@ -18,7 +18,7 @@ import traceback
 from tkinter import ttk
 from typing import TYPE_CHECKING, Any, Callable, Deque, Dict, Iterator, Optional, TextIO, Type, TypeVar, Union, cast
 
-import dacite  # type: ignore[import]
+import dacite
 
 if sys.version_info >= (3, 8):
     from typing import Literal
@@ -619,17 +619,17 @@ else:
     # written similarly to ttk.Combobox
     class Spinbox(ttk.Entry):
 
-        def __init__(self, master, *, from_=None, **kwargs):   # type: ignore[no-untyped-def]
+        def __init__(self, master: Any, *, from_: Any = None, **kwargs: Any):
             if from_ is not None:
                 kwargs['from'] = from_  # this actually works
             super().__init__(master, 'ttk::spinbox', **kwargs)
 
-        def configure(self, *args, **kwargs):   # type: ignore[no-untyped-def]
+        def configure(self, *args: Any, **kwargs: Any) -> Any:
             if 'from_' in kwargs:
                 kwargs['from'] = kwargs.pop('from_')
             return super().configure(*args, **kwargs)
 
-        config = configure   # type: ignore[assignment]
+        config = cast(Any, configure)
 
         if TYPE_CHECKING:
             def cget(self, key: str) -> Any: ...

--- a/requirements-dev.txt
+++ b/requirements-dev.txt
@@ -2,7 +2,7 @@ pytest
 isort>=5.7.0
 coverage
 sphinx
-mypy>=0.782
+mypy>=0.790
 pyflakes>=2.2.0
 pycodestyle>=2.6.0
 pytest-mock>=3.2.0,<4.0.0

